### ISSUE TITLE
Global Network Settings - KeepassXC integration

### ISSUE THREAD

**Issue by popxunga** [asbru-cm/asbru-cm#639]
Hi Team ...
Please consider the integration of Global Network Settings with KeepassXC.
As already implemented in other parts of Asbru, the idea is simply read the Host, Userid and Password from KeepassXC.
Thanks a lot.

**Comment by gfrenoy** [comment]
Made a first step ; let me know if this goes into the right direction...

**Comment by popxunga** [comment]
Hi @gfrenoy !
Yes I think so :-) Just tested it and it seems to works fine.
I really like the url, uid and pwd filled at once from the keepassxc lookup. I guess you should consider implement the same in the rest of the Asbru session settings panels.
By executing a **ps -ef | grep ssh**  can see the proxy credentials in clear which is not good.
Thanks.

**Comment by gfrenoy** [comment]
> Just tested it and it seems to works fine.

Thanks for the feedback :)

> I guess you should consider implement the same in the rest of the Asbru session settings panels.

Why not ... But where exactly ?

> By executing a ps -ef | grep ssh I can see the proxy credentials in clear which is not good.

Mmm, indeed does not look good ; we should create a new issue with an example so that I can see what we can do with this.

> Thanks.

Thanks to you !

**Comment by popxunga** [comment]
> > Just tested it and it seems to works fine.
> 
> Thanks for the feedback :)
> 
> > I guess you should consider implement the same in the rest of the Asbru session settings panels.
> 
> Why not ... But where exactly ?
Basically in every place where the integration with KeepassXC is in place which means while editing a connection:
-  in the Connection Details tab (the keepassxc button is there already but only the uid and pwd are read from keepassXC; the host/ip field is read individually);
- in the Network settings tab where there's no KeepassXC button and every url, uid and pwd field has to be read from keepassxc individually.
> 
> > By executing a ps -ef | grep ssh I can see the proxy credentials in clear which is not good.
> 
> Mmm, indeed does not look good ; we should create a new issue with an example so that I can see what we can do with this.
Issue #642 opened.
> 
> > Thanks.
> 
> Thanks to you !
You're welcome !

**Comment by gfrenoy** [comment]
> implement the same in the rest of the Asbru session settings panels.

Long time forgotten but I revisited this today and KeePassXC integration should be even better now :)

Let me know if this works for you.

And thanks again for all your great suggestions.

**Comment by popxunga** [comment]
Hi @gfrenoy !
Hope you're fine :-)
Finally I got some time to come back here.
And YESSS, I just tested the new loki build for awhile and it seems to work just fine :-D Thanks a lot !
I'm going to open a new issue for another little improvement regarding the integration with KeepassXC.
Basically is the error message displayed by Asbru when some of the values referenced on Asbru sessions that are used to lookup on KeepassXC get's deleted, moved, or renamed on KeepassXC.
Stay safe and cheers !
